woocommerce/woocommerce-gateway-stripe
# Add create test release script

# Changes proposed in this Pull Request:

This PR add a bash script to the bin folder, to automate the creation of a test release version.

Before doing executing any steps a process summary and a confirmation to continue will be displayed

```
This script will perform the following actions:
  - Create a new branch named 'release/5.6.0' using 'develop' as base, and push it to GitHub
  - Run Woorelease simulation locally with version=5.6.0 (using the 'release/5.6.0' branch)
  - Create a new tag named '5.6.0-test' in GitHub
  - Create a new release named '5.6.0-test' in GitHub (marked as 'pre-release')
  - Upload the generated plugin zip file (woocommerce-gateway-stripe.zip) to the newly created release in GitHub

Do you want to continue? (yes/no)
```

Next steps:
- Convert it to a javascript task
- Read the GitHub token from the Woorelease config

# Testing instructions

To test it without changing this repo, fork it and change the GITHUB_ACCOUNT variable at the top of the script.

__NOTE__: the script will be removed during the process (when switching branches), that is expected 😄 

- Execute the script with version number  5.9.0
  - `./bin/create-test-release.sh 5.9.0`

- Follow the script instructions
  - To test the automatic install of Woorelease a GitHub token with repo access is required, if left empty instructions to manually install it will be displayed
  - To test the automatic release creation and asset upload a GitHub token is required, if left empty instructions to manually create the release will be displayed

- Verify that the branch `release/5.9.0` was created
  - In this branch verify that both `readme.txt` and `changelog.txt` have the correct version number (5.x.x replaced with 5.9.0)... the date should remain as 2021-xx-xx as that will be changed by woorelease at the time of the production release.
  
- Verify that the tag `5.9.0-test` was created

- Extract the generated file `woocommerce-gateway-stripe.zip` and verify the version numbers in:
  - woocommerce-gateway-stripe.php:8
  - readme.txt:7,129
  - changelog.txt:3
 
#### Test cleanup:
- Delete the branch `release/5.9.0`
- Delete the tag `5.9.0-test`
